File: yh_tutorial/yh_tutorial_2/src/yh_sub_2_time.cpp
#include "ros/ros.h"
#include "yh_tutorial_2/yh_msg_2.h"

void msgCallback(const yh_tutorial_2::yh_msg_2::ConstPtr & msg)
{
    ROS_INFO("%d", msg->stamp.sec);
    ROS_INFO("%d", msg->stamp.nsec);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_sub_2_time");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_topic_2", 100, msgCallback);

    ros::spin();

    return 0;
}

File: yh_tutorial/yh_tutorial_2/scripts/yh_sub_2_time.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_2.msg import yh_msg_2

def msgCallback(msg):
    rospy.loginfo("%d", msg.stamp.secs)
    rospy.loginfo("%d", msg.stamp.nsecs)

def listener():
    rospy.init_node("yh_sub_2_time", anonymous=True)
    rospy.Subscriber("yh_topic_2", yh_msg_2, msgCallback)
    rospy.spin()

if __name__ == "__main__":
    listener()